File: 0389-find-the-difference/0389-find-the-difference.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
       vector<int>v(27); 
       for (auto i : s)
       {
           v[i-'a']++; 
       }
       
       for (auto i : t)
       {
           
            v[i-'a']--; 
       }
       for (int i=0;i<v.size();i++)
       {
           if (v[i]<0)
              return char('a'+i);
       }
       return 'a' ; 

    }
};

File: 0389-find-the-difference/0389-find-the-difference.cs
public class Solution {
    public char FindTheDifference(string s, string t) {
     Dictionary<char, int> dic = new Dictionary<char, int>(); 
            foreach (char c in s)
            {
               if(dic.ContainsKey(c))
               {
                    dic[c]++; 
               }
               else
                {
                    dic.Add(c, 1);
                }
            }
            foreach (char c in t)
            {
                if (dic.ContainsKey(c))
                {
                    dic[c]--; 
                }
                else
                {
                    dic.Add(c, -1);
                }
            }
            foreach (var item in dic)
            {
                if (item.Value<0)
                   return item.Key;
            }   
            return 'a'; 
    }
}